src/App.tsx

```diff
@@ -1,26 +1,158 @@
+        setErrorMessage(Error.LoadTodos);
+      }
+    })();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      if (filter === Filter.All) {
+        return true;
+      }
+
+      return filter === Filter.Completed ? todo.completed : !todo.completed;
+    });
+  }, [todos, filter]);
+
+  const todosCounter: number = useMemo(() => {
+    return todos.filter(todo => !todo.completed).length;
+  }, [todos]);
+
+  const todosCompletedCounter: number = useMemo(() => {
+    return todos.filter(todo => todo.completed).length;
+  }, [todos]);
+
+  const onAddTodo = async (todoTitle: string) => {
+    setTempTodo({ id: 0, title: todoTitle, completed: false, userId: USER_ID });
+    try {
+      const newTodo = await addTodo({ title: todoTitle, completed: false });
+
+      setTodos(prev => [...prev, newTodo]);
+    } catch (err) {
+      setErrorMessage(Error.AddTodos);
+      throw err;
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const onDeleteTodo = async (todoId: number) => {
+    setLoadingIds(prev => [...prev, todoId]);
+    try {
+      await deleteTodo(todoId);
+
+      setTodos(prev => prev.filter(todo => todo.id !== todoId));
+    } catch (err) {
+      setErrorMessage(Error.DeleteTodos);
+      throw err;
+    } finally {
+      setLoadingIds(prev => prev.filter(id => id !== todoId));
+    }
+  };
+
+  const onClearCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    completedTodos.forEach(todo => onDeleteTodo(todo.id));
+  };
+
+  // useEffect(() => {
+  //   setFooterVisible(visibleTodos.length > 0);
+  // }, [visibleTodos]);
+
+  // useEffect(() => {
+  //   setAreAllCompleted(visibleTodos.every(todo => todo.completed));
+  // }, [visibleTodos]);
+
+  // const handleToggleAllTodos = () => {
+  //   setVisibleTodos(prevTodos =>
+  //     prevTodos.map(todo => ({
+  //       ...todo,
+  //       completed: !areAllCompleted,
+  //     })),
+  //   );
+  // };
+
+  const handleToggleTodo = (todoId: number) => {
+    setTodos((prevTodos: Todo[]): Todo[] =>
+      prevTodos.map((todo: Todo) =>
+        todo.id === todoId ? { ...todo, completed: !todo.completed } : todo,
+      ),
+    );
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader
+          todos={todos}
+          onAddTodo={onAddTodo}
+          setErrorMessage={setErrorMessage}
+          isInputDisabled={!!tempTodo}
+        />
+
+        {todos.length > 0 && (
```

Comment:
it shoud be a good approach to using reduction to a boolean:
{!!todos.length && (code to render)}